src/components/ModalNew.tsx

```diff
@@ -0,0 +1,166 @@
+import { colorsV2 } from '@hedviginsurance/brand'
+import { motion } from 'framer-motion'
+import hexToRgba from 'hex-to-rgba'
+import * as React from 'react'
+import styled from 'react-emotion'
+import { Cross } from './icons/Cross'
+
+export interface ModalProps {
+  isVisible: boolean
+  onClose: () => void
+}
+
+const Wrapper = styled(motion.div)`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  left: 0;
+  top: 0;
+  bottom: 0;
+  right: 0;
+  z-index: 2000;
+`
+
+const Background = styled(motion.div)`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  left: 0;
+  top: 0;
+  bottom: 0;
+  right: 0;
+  background-color: ${hexToRgba(colorsV2.white, 0.75)};
+`
+
+const Container = styled(motion.div)`
+  position: relative;
+  width: 100%;
+  max-width: 900px;
+  height: 100%;
+  min-height: 400px;
+  max-height: 900px;
+  background: ${colorsV2.white};
+  border-radius: 9px;
+  position: absolute;
+  left: 50%;
+  top: 50%;
+  transform: translateX(-50%) translateY(-50%);
+  box-shadow: 0 0 14px rgba(0, 0, 0, 0.06);
+  box-sizing: border-box;
+  overflow-x: scroll;
+
+  @media (max-height: 900px) {
+    max-height: calc(100vh - 32px);
+  }
+
+  @media (max-width: 900px) {
+    max-width: calc(100% - 32px);
+  }
+`
+
+const CloseButton = styled('button')`
+  width: 24px;
+  height: 24px;
+  position: absolute;
+  top: 16px;
+  right: 16px;
+  padding: 0;
+  display: flex;
+  align-items: center;
+  text-align: center;
+  background-color: ${colorsV2.gray};
+  border-radius: 50%;
+  border: none;
+  cursor: pointer;
+
+  :focus {
+    outline: none;
+  }
+
+  :hover {
+    background-color: ${colorsV2.darkgray};
+  }
+
+  svg {
+    width: 100%;
+    height: 100%;
+    fill: ${colorsV2.white};
+  }
+`
+
+export const Modal = (props: React.PropsWithChildren<ModalProps>) => {
+  const containerRef = React.useRef(null)
```

Comment:
We should probably pass the type to `useRef` so that this becomes `React.useRef<HTMLDivElement>(null)`  - that way we don't need to disable ts in some places i think